fix: do not crash if distribution cannot be found when extracting semver

hannahrogers-google opened the pull request:
In the case where a user is not running from within a pip package, the google-cloud-pubsub distribution might not be available. At the moment this will cause a fatal DistributionNotFound error because in the publisher and subscriber client, we try to extract the distribution version using 'pkg_resources.get_distribution("google-cloud-pubsub").version'.

In the case where a distribution is not found, making the change to set the version to 0.0.

pradn reviewed (changes requested):
Please add some context for when this call can throw.

pradn commented:
Change looks good! Can you please update the PR summary with some context?

hannahrogers-google commented:
> Change looks good! Can you please update the PR summary with some context?

Yes, I updated the PR summary and added a doc for context in the PR.  Thanks, Prad!

pradn commented:
Looks good, approved!
